lib/filewatch/watch.rb

```diff
@@ -118,13 +118,21 @@ def each(&block)
       synchronized do
         return if @files.empty?
 
+        quitting = false
         file_deleteable = []
         # creates this array just once
         watched_files = @files.values
 
+        quit_proc = lambda do
```

Comment:
One thing, do you think the ensure block executes in the synchronised block scope? I am not sure more than likely not.